### PR TITLE
chore: Clean up old responsibilities and mainResearchInterests

### PR DESCRIPTION
**Pull request by peterstarling**
https://trello.com/c/cJ2al5LQ/1731-2-clean-up-responsibilities-and-research-interests

### CODE REVIEW

File: packages/react-components/src/templates/__tests__/UserProfileResearch.test.tsx [hunk 1]
```diff
@@ -115,7 +112,7 @@ it('renders an edit button for the role on the team (REGRESSION)', () => {
           id: '42',
           displayName: 'Team',
           role: 'Lead PI (Core Leadership)',
-          editHref: '/edit-team-membership/42',
+          editHref: '/edit-role/42',
```
Comment (jhonny-m) [inline packages/react-components/src/templates/__tests__/UserProfileResearch.test.tsx:115]: this is no longer associated to the team, there is only one edit-role link
we can remove this line from here

File: apps/frontend/src/network/users/__tests__/UserProfile.test.tsx [hunk 2]
```diff
@@ -184,7 +183,6 @@ it("links to the user's team", async () => {
 });
 
 it("links to the user's team proposal", async () => {
-  disable('UPDATED_ROLE_SECTION');
```
Comment (jhonny-m) [inline apps/frontend/src/network/users/__tests__/UserProfile.test.tsx:187]: we can remove this whole test and the one right bellow
`'does not show a proposal for a user whose team has none'`

**Review by niezgoda — APPROVED** [review]

File: apps/asap-server/src/migrations/1639476045046-move-responsabilities-interests-from-teams-to-users.ts [hunk 3]
```diff
@@ -11,15 +12,19 @@ export default class AddDefaultOnboardedFieldsToUser extends Migration {
         await squidexClient.patch(user.id, {
           researchInterests: {
             iv:
-              user.data.teams.iv?.find(
-                (team) => !!team.mainResearchInterests?.length,
-              )?.mainResearchInterests ?? '',
+              (
+                user.data.teams.iv?.find(
+                  (team) => !!(team as any).mainResearchInterests?.length,
+                ) as any
```
Comment (kmaid) [inline apps/asap-server/src/migrations/1639476045046-move-responsabilities-interests-from-teams-to-users.ts:18]: This is irritating to have to do

File: apps/frontend/src/network/users/__tests__/Research.test.tsx [hunk 4]
```diff
@@ -87,75 +86,54 @@ describe('when editing', () => {
...
     ],
   };
 
   let result!: RenderResult;
   beforeEach(async () => {
-    disable('UPDATED_ROLE_SECTION');
     result = render(<Research user={user} />, { wrapper });
     await result.findAllByLabelText(/edit/i);
   });
   describe('team membership', () => {
```
Comment (kmaid) [inline apps/frontend/src/network/users/__tests__/Research.test.tsx:106]: This should be updated

**Review by kmaid — APPROVED** [review]

**Pull request merged**